modules/gapi/test/common/gapi_imgproc_tests_common.hpp

```diff
@@ -14,6 +14,105 @@
+    }
+}
+
+enum WithHierarchy {NO_HIERARCHY, WITH_HIERARCHY};
+
+template<WithHierarchy withHierarchy = NO_HIERARCHY>
+cv::GComputation findContoursTestGAPI(const cv::Mat& in, const cv::RetrievalModes mode,
+                                      const cv::ContourApproximationModes method,
+                                      cv::GCompileArgs&& args,
+                                      std::vector<std::vector<cv::Point>>& out_cnts_gapi,
+                                      std::vector<cv::Vec4i>&,
+                                      const cv::Point& offset = cv::Point())
+{
+    cv::GMat g_in;
+    cv::GOpaque<cv::Point> gOffset;
+    cv::GArray<cv::GArray<cv::Point>> outCts;
+    outCts = cv::gapi::findContours(g_in, mode, method, gOffset);
+    cv::GComputation c(GIn(g_in, gOffset), GOut(outCts));
+    c.apply(gin(in, offset), gout(out_cnts_gapi), std::move(args));
+    return c;
+}
+
+template<> cv::GComputation findContoursTestGAPI<WITH_HIERARCHY> (
+    const cv::Mat& in, const cv::RetrievalModes mode, const cv::ContourApproximationModes method,
+    cv::GCompileArgs&& args, std::vector<std::vector<cv::Point>>& out_cnts_gapi,
+    std::vector<cv::Vec4i>& out_hier_gapi, const cv::Point& offset)
+{
+    cv::GMat g_in;
+    cv::GOpaque<cv::Point> gOffset;
+    cv::GArray<cv::GArray<cv::Point>> outCts;
+    cv::GArray<cv::Vec4i> outHier;
+    std::tie(outCts, outHier) = cv::gapi::findContoursH(g_in, mode, method, gOffset);
+    cv::GComputation c(GIn(g_in, gOffset), GOut(outCts, outHier));
+    c.apply(gin(in, offset), gout(out_cnts_gapi, out_hier_gapi), std::move(args));
+    return c;
+}
+
+template<WithHierarchy withHierarchy = NO_HIERARCHY>
+void findContoursTestOpenCVCompare(const cv::Mat& in, const cv::RetrievalModes mode,
+                                   const cv::ContourApproximationModes method,
+                                   const std::vector<std::vector<cv::Point>>& out_cnts_gapi,
+                                   const std::vector<cv::Vec4i>&              out_hier_gapi,
+                                   const CompareMats& cmpF, const cv::Point& offset = cv::Point())
+{
+    // OpenCV code /////////////////////////////////////////////////////////////
+    std::vector<std::vector<cv::Point>> out_cnts_ocv;
+    std::vector<cv::Vec4i>              out_hier_ocv;
+    cv::findContours(in, out_cnts_ocv, out_hier_ocv, mode, method, offset);
+    // Comparison //////////////////////////////////////////////////////////////
+    EXPECT_TRUE(out_cnts_gapi.size() == out_cnts_ocv.size());
+
+    cv::Mat out_mat_ocv  = cv::Mat(cv::Size{ in.cols, in.rows }, in.type(), cv::Scalar::all(0));
+    cv::Mat out_mat_gapi = cv::Mat(cv::Size{ in.cols, in.rows }, in.type(), cv::Scalar::all(0));
+    cv::fillPoly(out_mat_ocv,  out_cnts_ocv,  cv::Scalar::all(1));
+    cv::fillPoly(out_mat_gapi, out_cnts_gapi, cv::Scalar::all(1));
+    EXPECT_TRUE(cmpF(out_mat_ocv, out_mat_gapi));
+    if (withHierarchy == WITH_HIERARCHY)
```

Comment:
`std::is_same<withHierarchy, WITH_HIERARCHY>::value`. M.B.

Reply:
But. Leave it as it is.

Reply:
Yeah, I think it is not better, too